src/test/java/com/thebois/models/beings/ColonyTests.java

```diff
@@ -49,4 +61,98 @@ public void updateCallsUpdateOnAllBeings() {
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("getItemTypes")
+    public void emptyColonyInventoryIsEmpty(final ItemType itemType) {
+        // Arrange
+        final int beingCount = 1;
+        final Collection<IBeing> pawns = new ArrayList<>(beingCount);
+        for (int i = 0; i < beingCount; i++) {
+            pawns.add(Mockito.mock(IBeing.class));
+        }
+        final Colony colony = new Colony(pawns);
+
+        // Act
+        final Optional<IItem> result = colony.takeItemFromColonyInventory(itemType);
+
+        // Assert
+        assertThat(result.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void canAddAndTakeItemToInventory() {
+        // Arrange
+        final int beingCount = 1;
+        final Collection<IBeing> pawns = new ArrayList<>(beingCount);
+        for (int i = 0; i < beingCount; i++) {
+            pawns.add(Mockito.mock(IBeing.class));
+        }
+        final Colony colony = new Colony(pawns);
+
+        // Act
+        colony.addItemToColonyInventory(new Log());
+        final Optional<IItem> item = colony.takeItemFromColonyInventory(ItemType.LOG);
+
+        // Assert
+        assertThat(item.isPresent()).isTrue();
+        assertThat(item.get().getType()).isEqualTo(ItemType.LOG);
+    }
+
+    @Test
+    public void testCountEmptyInventory() {
+        // Arrange
+        final int beingCount = 1;
+        final Collection<IBeing> pawns = new ArrayList<>(beingCount);
+        for (int i = 0; i < beingCount; i++) {
+            pawns.add(Mockito.mock(IBeing.class));
+        }
+        final Colony colony = new Colony(pawns);
+
+        // Act
+        final int count = colony.getItemCount(ItemType.ROCK);
+
+        // Assert
+        assertThat(count).isEqualTo(0);
+    }
+
+    @Test
+    public void testCountIsNotZeroWhenInventoryGotSpecifiedItemTypeInIt() {
```

Comment:
style

No need to prefix it with `test`. And reading the test, it doesn't seem you check for count not being zero, instead you check for it being the correct value.

Reply:
Yup and yup 👍